amoro-common/src/main/java/org/apache/amoro/spi/PasswdAuthenticationProvider.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.amoro.spi;
+
+import org.apache.amoro.exception.SignatureCheckException;
+
+public interface PasswdAuthenticationProvider {
+  /**
+   * The authenticate method is called by the amoro Server authentication layer to authenticate
+   * users for their requests. If a user is to be granted, return nothing/throw nothing. When a user
+   * is to be disallowed, throw an appropriate [[SignatureCheckException]].
+   *
+   * @param user The username received over the connection request
+   * @param password The password received over the connection request
+   * @throws SignatureCheckException When a user is found to be invalid by the implementation
+   */
+  void authenticate(String user, String password) throws SignatureCheckException;
```

Comment:
the return value should be a `Principal` so you can carry more information for the authenticated user instead of a simple yes or no.

You can refer to Trino's `io.trino.spi.security.PasswordAuthenticator`